src/libmongoc/src/mongoc/mongoc-change-stream.c

```diff
@@ -73,6 +73,7 @@ _is_resumable_error (mongoc_change_stream_t *stream, const bson_t *reply)
    case MONGOC_SERVER_ERR_STALECONFIG:
    case MONGOC_SERVER_ERR_RETRYCHANGESTREAM:
    case MONGOC_SERVER_ERR_FAILEDTOSATISFYREADPREFERENCE:
+   case MONGOC_SERVER_ERR_LEGACYNOTPRIMARY:
```

Comment:
Not sure if this is mandated by the spec, but I added this error code to all places where we check for the other two errors that indicate that the server is not a primary.

Reply:
On second thought, I've removed this as it does not seem intentional per the spec text (and probably does not apply since the error code was used pre-3.2)